classes/db/Db.php

```diff
@@ -790,53 +854,83 @@ public function escape($string, $html_ok = false, $bq_sql = false)
      */
     public static function checkConnection($server, $user, $pwd, $db, $new_db_link = true, $engine = null, $timeout = 5)
     {
-        return call_user_func_array(array(Db::getClass(), 'tryToConnect'), array($server, $user, $pwd, $db, $new_db_link, $engine, $timeout));
+        return self::tryToConnect($server, $user, $pwd, $db);
     }
 
     /**
-     * Try a connection to the database and set names to UTF-8
+     * Try a connection to the database
      *
+     * @see Db::checkConnection()
      * @param string $server Server address
      * @param string $user Login for database connection
      * @param string $pwd Password for database connection
-     * @return bool
+     * @param string $db Database name
+     * @param bool $newDbLink
+     * @param string|bool $engine
+     * @param int $timeout
+     * @return int Error code or 0 if connection was successful
      */
-    public static function checkEncoding($server, $user, $pwd)
+    public static function tryToConnect($server, $user, $pwd, $db, $new_db_link = true, $engine = null, $timeout = 5)
     {
-        return call_user_func_array(array(Db::getClass(), 'tryUTF8'), array($server, $user, $pwd));
+        try {
+            $link = new self($server, $user, $pwd, $db, true);
+        } catch (DBALException $e) {
+            return ($e->getCode() == 1049 || (defined('HHVM_VERSION') && $e->getCode() == 42000)) ? 2 : 1;
+        }
+        unset($link);
+        return 0;
     }
-
     /**
-     * Try a connection to the database and check if at least one table with same prefix exists
+     * Selects best table engine.
      *
-     * @param string $server Server address
-     * @param string $user Login for database connection
-     * @param string $pwd Password for database connection
-     * @param string $db Database name
-     * @param string $prefix Tables prefix
-     * @return bool
+     * @return string
      */
-    public static function hasTableWithSamePrefix($server, $user, $pwd, $db, $prefix)
+    public function getBestEngine()
     {
-        return call_user_func_array(array(Db::getClass(), 'hasTableWithSamePrefix'), array($server, $user, $pwd, $db, $prefix));
+        $value = 'InnoDB';
+        $sql = 'SHOW VARIABLES WHERE Variable_name = \'have_innodb\'';
+        $result = $this->link->query($sql);
+        if (!$result) {
+            $value = 'MyISAM';
+        }else {
+            $row = $result->fetch();
+            if (!$row || strtolower($row['Value']) != 'yes') {
+                $value = 'MyISAM';
+            }
+        }
+        /* MySQL >= 5.6 */
+        $sql = 'SHOW ENGINES';
```

Comment:
FYI SELECT ENGINE FROM INFORMATION_SCHEMA.ENGINES; covers MySQL >= 5.0.
Not needed anymore if SHOW VARIABLES LIKE 'innodb_%'; is used